pkg/services/certgenerator/certutil.go

```diff
@@ -0,0 +1,185 @@
+package certgenerator
+
+import (
+	"bytes"
+	cryptorand "crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"time"
+
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	"k8s.io/kubernetes/pkg/controlplane"
+	netutils "k8s.io/utils/net"
+	"net"
+	"strings"
+)
+
+const (
+	CACertFile = "data/k8s/ca.crt"
+	CAKeyFile  = "data/k8s/ca.key"
+
+	APIServerCertFile = "data/k8s/apiserver.crt"
+	APIServerKeyFile  = "data/k8s/apiserver.key"
+)
+
+type CertUtil struct {
+	caKey  *rsa.PrivateKey
+	caCert *x509.Certificate
+}
+
+// Lifted from apiserver package as it's a dependency of external cert generation
```

Comment:
could you add code comments with links to the source of the copied code/functions, and maybe also comments around any places that needed to be modified for our use case?